File: app/src/main/java/de/christinecoenen/code/zapp/app/player/IPlaybackPositionRepository.java
package de.christinecoenen.code.zapp.app.player;

import androidx.annotation.NonNull;

import io.reactivex.Single;

public interface IPlaybackPositionRepository {

	void savePlaybackPosition(@NonNull VideoInfo videoInfo, long positionMillis, long durationMillis);

	Single<Long> getPlaybackPosition(@NonNull VideoInfo videoInfo);

}

File: app/src/main/java/de/christinecoenen/code/zapp/app/player/IPlaybackPositionRepository.kt
package de.christinecoenen.code.zapp.app.player

interface IPlaybackPositionRepository {

	suspend fun savePlaybackPosition(
		videoInfo: VideoInfo,
		positionMillis: Long,
		durationMillis: Long
	)

	suspend fun getPlaybackPosition(videoInfo: VideoInfo): Long

}
